File: frontend/lib/types/superadmin.ts
// Tipi TypeScript allineati ai contratti Pydantic esposti da
// backend/app/api/superadmin.py.

export interface Consulente {
  id: string;
  nome: string;
  cognome: string;
  email: string;
}

export interface AziendaAmministrazione {
  id: string;
  ragione_sociale: string;
  partita_iva: string | null;
  codice_fiscale: string | null;
  stato_approvazione: "in_attesa" | "approvata" | "rifiutata";
  created_at: string;
  // Di norma uno solo (chi l'ha creata), ma può averne più d'uno.
  consulenti: Consulente[];
}

export interface AziendaSintesi {
  id: string;
  ragione_sociale: string;
  stato_approvazione: "in_attesa" | "approvata" | "rifiutata";
}

export interface ConsulenteAmministrazione extends Consulente {
  attivo: boolean;
  aziende: AziendaSintesi[];
}

File: backend/app/schemas/superadmin.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ConsulenteRead(BaseModel):
    id: UUID
    nome: str
    cognome: str
    email: str


class AziendaAmministrazioneRead(BaseModel):
    id: UUID
    ragione_sociale: str
    partita_iva: str | None
    codice_fiscale: str | None
    stato_approvazione: str
    created_at: datetime
    # Di norma uno solo (chi l'ha creata): una lista perché un'azienda può
    # avere più consulenti assegnati (vedi POST .../consulenti).
    consulenti: list[ConsulenteRead] = []


class AssociaConsulenteRequest(BaseModel):
    consulente_id: UUID
